std/format.d

```diff
@@ -4359,7 +4359,7 @@ T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     enforce(spec.spec == 's',
             text("Wrong unformat specifier '%", spec.spec , "' for ", T.stringof));
 
-    return parse!T(input);
+    static if(__traits(compiles, parse!T(input)) ) return parse!T(input);
```

Comment:
This doesn't work. In case the static test fails, the function has no return value and fails to compile. Phobos functions should not fail to compile (only refuse to accept types depending on constraints). So one possibility is to move the condition in the constraint like this:

```d
T unformatValue(T, Range, Char)(ref Range input, ref FormatSpec!Char spec)
     if (isInputRange!Range && is(T == typeof(null)) && is(typeof(parse!T(input)) == T))
{
     import std.conv : parse, text;
      enforce(spec.spec == 's',
              text("Wrong unformat specifier '%", spec.spec , "' for ", T.stringof));
     return parse!T(input);
}
```

However, this is a very specialized function only parsing "null" so I assume you don't even need the extra check.